### PR TITLE
Cut, Pour and Storage related dispositions 

### PR DESCRIPTION
**Pull request by AbhijitVyas**

### CODE REVIEW

File: owl/SOMA-OBJ.owl
Comment (kaviyachandran) [inline owl/SOMA-OBJ.owl:1422]: I defined [Cuttability](https://github.com/kaviyachandran/situational_awareness_example/blob/master/owl/situation_awareness_example.owl#L711) as a disposition property of the objects that take the role of CutObject

File: owl/SOMA-OBJ.owl
Comment (kaviyachandran) [inline owl/SOMA-OBJ.owl:1443]: You could directly use CuttingTool

File: owl/SOMA-OBJ.owl
Comment (kaviyachandran) [inline owl/SOMA-OBJ.owl:1709]: You can just use Containment

**Comment by mpomarlan** [comment]
Having looked through this, I can summarize my review this way:

1) Kaviya's observation about overlaps with concepts from SOMA-HOME is a good one, and we should decide what to do about this redundancy.

2) I may need a refresher into how the affordance model of SOMA developped :P Here is my understanding of it so far, and correct me if I am wrong here. An affordance is a relation between dispositions of objects. If it can be said to describe a disposition, it describes all dispositions taking part in it. So, for example, an affordance about cutting describes both the disposition of an object to be cuttable, and the disposition of another object to cut others.

One of these dispositions is the "bearer", the other the "trigger". The difference is the following: the bearer is the one "acted upon", or more clearly, the bearer is the entity undergoing some significant process or change. The trigger may change too if you look at things in detail, but if you conceptualize something as a "trigger", you're saying you don't care about, as in, you abstract away from, any change in the trigger. Once a knife cuts a stick of butter it is still the same knife. Once the stick of butter gets cut, well you have to account for some different entities that have appeared etc.

There are complications using this heuristic of bearer vs trigger (what happens if the trigger also gets used up, such as when using a match to light a stove?) so here is another one, perhaps more robust: the bearer is the entity which undergoes some relevant change as a result of the affordance manifesting. Relevant means aligned or counter-aligned with someone's goals. As in, when I light the stove with a match, I do this because I intend the stove to be on fire; the fact that the match also gets burnt is incidental, it is neither an intention of mine nor do I intend the match to not be burnt.

To bring this round to the pull request here, I look at an axiom saying that CanCut affordsBearer CuttingTool with suspicion. CuttingTool is not a bearer role in an affordance about cutting, it is a trigger role. I see something similar happening for ShapedObject in an already accepted part of the ontology.

My suggestion -- can we discuss this in the onto meeting? I want to hear what the meaning of affordances is now, and why that changed.

File: owl/SOMA-OBJ.owl
Comment (AbhijitVyas) [reply owl/SOMA-OBJ.owl:1407]: As far as I see in SOMA-HOME.owl, a CuttingTool is of type DesignedTool (is a PhysicalObject) which has some taxonomy underneath such as Knife etc. My proposal of new concept called Cutter is not of type PO but rather a role which is assumed by PO such as a knife during a cutting event. I see dispositional axioms such as 'affordsBearer' and 'affordsTrigger' associated with Role and hence the concept of 'Cutter'.  We can for sure discuss this thing during over meeting today.

File: owl/SOMA-OBJ.owl
Comment (AbhijitVyas) [reply owl/SOMA-OBJ.owl:1443]: See my comment above.

File: owl/SOMA-OBJ.owl
Comment (AbhijitVyas) [reply owl/SOMA-OBJ.owl:1709]: yes but it is far more than a Containment. For example; a spoon contains a liquid but it can not store it. Hence a need for a new dispositional concept. :)

Comment (kaviyachandran) [reply owl/SOMA-OBJ.owl:1709]: Yes, as Mihai mentioned, there needs to be a distinction between containment and storage

**Comment by daniel86** [comment]
I removed myself and Sascha as reviewer. 4 reviews are not needed. @AbhijitVyas you should have enough input for updating the pull request now, or?

File: owl/SOMA-OBJ.owl
Comment (AbhijitVyas) [reply owl/SOMA-OBJ.owl:1474]: @mpomarlan and @kaviyachandran , do we agree upon this discussion here? If yes I would like to resolve this conversation.

**Comment by AbhijitVyas** [comment]
> Having looked through this, I can summarize my review this way:
> 
> 1. Kaviya's observation about overlaps with concepts from SOMA-HOME is a good one, and we should decide what to do about this redundancy.
> 2. I may need a refresher into how the affordance model of SOMA developped :P Here is my understanding of it so far, and correct me if I am wrong here. An affordance is a relation between dispositions of objects. If it can be said to describe a disposition, it describes all dispositions taking part in it. So, for example, an affordance about cutting describes both the disposition of an object to be cuttable, and the disposition of another object to cut others.
> 
> One of these dispositions is the "bearer", the other the "trigger". The difference is the following: the bearer is the one "acted upon", or more clearly, the bearer is the entity undergoing some significant process or change. The trigger may change too if you look at things in detail, but if you conceptualize something as a "trigger", you're saying you don't care about, as in, you abstract away from, any change in the trigger. Once a knife cuts a stick of butter it is still the same knife. Once the stick of butter gets cut, well you have to account for some different entities that have appeared etc.
> 
> There are complications using this heuristic of bearer vs trigger (what happens if the trigger also gets used up, such as when using a match to light a stove?) so here is another one, perhaps more robust: the bearer is the entity which undergoes some relevant change as a result of the affordance manifesting. Relevant means aligned or counter-aligned with someone's goals. As in, when I light the stove with a match, I do this because I intend the stove to be on fire; the fact that the match also gets burnt is incidental, it is neither an intention of mine nor do I intend the match to not be burnt.
> 
> To bring this round to the pull request here, I look at an axiom saying that CanCut affordsBearer CuttingTool with suspicion. CuttingTool is not a bearer role in an affordance about cutting, it is a trigger role. I see something similar happening for ShapedObject in an already accepted part of the ontology.
> 
> My suggestion -- can we discuss this in the onto meeting? I want to hear what the meaning of affordances is now, and why that changed.

We should discuss this PR in tomorrow's ontology meeting one last time in order to remove any confusion.

**Comment by mpomarlan** [comment]
I see the branch has a few conflicts that make github unable to merge. I'll start the review when these are fixed or next week Wednesday, whichever comes later.

**Comment by AbhijitVyas** [comment]
> I see the branch has a few conflicts that make github unable to merge. I'll start the review when these are fixed or next week Wednesday, whichever comes later.

Done. The conflicts are resolved.

**Comment by mpomarlan** [comment]
Excellent!

File: owl/SOMA-OBJ.owl
Comment (kaviyachandran) [reply owl/SOMA-OBJ.owl]: > 

Does it make sense to say the objects which can be used to store has both dispositions containment and blockage?

Comment (AbhijitVyas) [reply owl/SOMA-OBJ.owl]: The same object can be used for multiple objectives and hence can poses many dispositions. But here the comment is leading to the direction of explaining how the container can be used to store things, and by doing so it can indirectly block free movements of things. However, new comment should not lead to such confusion.

**Review by mpomarlan — CHANGES_REQUESTED** [review]
Almost done :)

I noticed PouredObjecct has a typo, and while minor it would be nice to fix before merging.

A little more significant is the problem with Pourable. This actually is the fault of the English language, as we tend to use the same syntactic forms for very different things: "Alice poured the milk into the bowl" vs "Alice poured the cup into the bowl".

What this means, first, is that the concept names are not suggestive. Pourable could as well refer to the liquid being poured, and not the container it comes from (and vice-versa for PouredObject). Maybe there is a way to have more suggestive concept names.

We can live with somewhat ambiguous concept names if the concept comments are hard to misinterpret. This actually affects the comments for both Pourable and PouredObject.

E.g., PouredObject: Alice pours the cup onto the table. So will the PouredObject refer to the cup, or to the liquid that was in it?

E.g., Pourable: the first sentence of the comment is ambiguous (see Alice above). The comment is further strange in that it seems to want to resolve this ambiguity, but does so in conflicting ways. On the one hand, we have "(the barrier)" in the first sentence which suggests we are talking about the source container. However the second sentence talks about the fluid.

I suggest that, since natural language use is a little ambiguous when pouring is concerned, it will be ok if the comments will be a little "strained" in terms of expression. E.g. something like the disposition of a fluid or substance which makes it possible to pour it out of a container and into or onto other objects, a role of some fluid or substance that is the patient of pouring task. Still not quite perfect as not only fluids are pourable (powder is pourable and I'm not sure it counts as a fluid).

File: owl/SOMA-OBJ.owl
Comment (kaviyachandran) [reply owl/SOMA-OBJ.owl]: I will just check with @mpomarlan once today in the meeting. Yes, we can add the disposition as an enabler of common everyday usage. We cant add all the possible dispositions. I did not mean it in that way.

**Comment by AbhijitVyas** [comment]
> Almost done :)
> 
> I noticed PouredObjecct has a typo, and while minor it would be nice to fix before merging.
> 
> A little more significant is the problem with Pourable. This actually is the fault of the English language, as we tend to use the same syntactic forms for very different things: "Alice poured the milk into the bowl" vs "Alice poured the cup into the bowl".
> 
> What this means, first, is that the concept names are not suggestive. Pourable could as well refer to the liquid being poured, and not the container it comes from (and vice-versa for PouredObject). Maybe there is a way to have more suggestive concept names.
> 
> We can live with somewhat ambiguous concept names if the concept comments are hard to misinterpret. This actually affects the comments for both Pourable and PouredObject.
> 
> E.g., PouredObject: Alice pours the cup onto the table. So will the PouredObject refer to the cup, or to the liquid that was in it?
> 
> E.g., Pourable: the first sentence of the comment is ambiguous (see Alice above). The comment is further strange in that it seems to want to resolve this ambiguity, but does so in conflicting ways. On the one hand, we have "(the barrier)" in the first sentence which suggests we are talking about the source container. However the second sentence talks about the fluid.
> 
> I suggest that, since natural language use is a little ambiguous when pouring is concerned, it will be ok if the comments will be a little "strained" in terms of expression. E.g. something like the disposition of a fluid or substance which makes it possible to pour it out of a container and into or onto other objects, a role of some fluid or substance that is the patient of pouring task. Still not quite perfect as not only fluids are pourable (powder is pourable and I'm not sure it counts as a fluid).

Okay, the typo is removed and the comment is updated.

**Comment by mpomarlan** [comment]
This part ", a role of some fluid or substance that is the patient of pouring task" was not intended to be in the comment for the Pourable disposition but in the comment for the role you defined (PouredObject).

Otherwise good. Change the above and I will approve&merge.

**Comment by AbhijitVyas** [comment]
> This part ", a role of some fluid or substance that is the patient of pouring task" was not intended to be in the comment for the Pourable disposition but in the comment for the role you defined (PouredObject).
> 
> Otherwise good. Change the above and I will approve&merge.

Done.

**Review by mpomarlan — APPROVED** [review]

**Pull request merged**
